packages/typegpu/src/data/vector.ts

```diff
@@ -58,11 +73,7 @@ function makeVecSchema<TType extends string, TValue>(
       return `${VecSchema.type}(${values.join(', ')})` as unknown as TValue;
     }
 
-    if (values.length <= 1) {
-      return options.makeFromScalar(values[0] ?? 0);
-    }
-
-    if (values.length === options.length) {
+    if (values.length <= 1 || values.length === options.length) {
       return options.make(...values);
```

Comment:
Arguably, we could pass the values directly to the constructor of the vector and not check lengths. In case someone generates thousands of vectors, these checks can add up. We could simply rely on typescript to handle this for us and not pay the runtime cost (even if small).

I think having the option to initialize using `vec4(vec3(), w)` would be super cool, but I did not find a way to do this without paying the cost of checking args. If someone wants to do this in js, the simplest way would be to `vec4(...vec3(), w)`. This would work just fine, we would just need to adjust the transpiler to ignore the dots.